returnn/datasets/sprint.py

```diff
@@ -64,9 +64,12 @@ def __init__(self, target_maps=None, str_add_final_zero=False, input_stddev=1.,
     :param None|dict[str] bpe: if given, will be opts for :class:`BytePairEncoding`
     :param None|dict[str] orth_vocab: if given, orth_vocab is applied to orth and orth_classes is an available target`
     :param bool suppress_load_seqs_print: less verbose
+    :param int reduce_target_factor: downsample factor to allow less targets than features
```

Comment:
What about allowing arbitrary floats >0 (I.e. Also upsampling)?

Reply:
In that case the ceildiv would not work properly. But maybe it would be good to allow an arbitrary callable. Then the user could manually specify how the sequence length is changed. This would also cover the downsample->upsample->downsample case you mentioned before.

Reply:
Yes but I think I would only add that once we really need it.

Reply:
It sure would still work. E.g. `ceildiv(100/1.8) = 55` or  `ceildiv(100/0.33) = 303`

Reply:
> It sure would still work. E.g. `ceildiv(100/1.8) = 55` or `ceildiv(100/0.33) = 303`

That does not make sense. Maybe you mean `math.ceil` here? But using `math.ceil` is problematic. There are much better ways to implement `ceildiv`, see [here](https://stackoverflow.com/questions/14822184/is-there-a-ceiling-equivalent-of-operator-in-python). The current implementation in the PR is also fine. But eg. this implementation would not work (as-is) for float, because it would return not an int.

Anyway, I think we can just leave it like that for now. I think I would only extend it once we really need it, and then maybe directly go with the callable for full flexibility.

Reply:
Ok from my side